typed-clojure/src/typed_clojure/main.clj

```diff
@@ -0,0 +1,280 @@
+  (if (or (string? x) (number? x))
+    (if (or (number? x) (boolean? x))
+      (t/ann-form (count x) t/Num) ; Type error: x could be number
+      0)
+    0))
+
+;; Example struct_fields
+;; success
+(t/defalias MyStruct
+  (t/HMap :mandatory {:a (t/U nil Number)}
+          :complete? true))
+(t/ann struct-fields-success-f [MyStruct -> t/Num])
+(defn struct-fields-success-f [x]
+  (if (number? (:a x))
+    (:a x)
+    0))
+
+;; failure
+(t/defalias MyStruct
+  (t/HMap :mandatory {:a (t/U nil Number)}
+          :complete? true))
+(t/ann struct-fields-failure-f [MyStruct -> t/Num])
+(defn struct-fields-failure-f [x]
+  (if (string? (:a x))
+    (t/ann-form (:a x) t/Num) ; Error: can't cast string to Num
+    0))
+
+;; Example tuple_elements
+;; success
+(t/ann tuple-elements-success-f [(t/NonEmptyVec (t/U nil t/Num)) :-> t/Num])
+(defn tuple-elements-success-f [x]
+  (let [first-elem (nth x 0)]
+    (if (number? first-elem)
+      first-elem
+      0)))
+
+;; failure
+(t/ann tuple-elements-failure-f [(t/NonEmptyVec (t/U nil t/Num)) :-> t/Num])
+(defn tuple-elements-failure-f [x]
+  (if (number? (nth x 0))
+    (+ (nth x 0) (nth x 1)) ; Type error: (nth x 1) could be non-number
+    0))
+
+;; Example tuple_length
+;; success
+(t/ann tuple-length-success-f [(t/NonEmptyVec t/Num) :-> t/Num])
+(defn tuple-length-success-f [x]
+  (if (= (count x) 2)
+    (let [first-elem (nth x 0)
+          second-elem (nth x 1)]
+      (+ first-elem second-elem))
+    0))
+
+;; failure
+(t/ann tuple-length-failure-f [(t/U (t/NonEmptyVec t/Num) (t/NonEmptyVec t/Str)) :-> t/Num])
+(defn tuple-length-failure-f [x]
+  (if (= (count x) 2)
+    (+ (nth x 0) (nth x 1))
+    (+ (nth x 0) (nth x 1)))) ; Type error: with expected type: t/Num
+
+;; Example alias
+;; success
+(t/ann alias-success-f [t/Any :-> (t/U t/Int t/Any)])
+(defn alias-success-f [x]
+  (let [y (string? x)]
+    (if y
+      (count x)
+      x)))
+
+;; failure
+(t/ann alias-failure-f [t/Any :-> (t/U t/Int t/Any)])
+(defn alias-failure-f [x]
+  (let [y (string? x)]
+    (if y
+      (t/ann-form (.isNaN x) t/Num) ; Type error: strings don't have .isNaN
+      x)))
+
+(t/ann alias-failure-g [t/Any :-> (t/U t/Int t/Any)])
+(defn alias-failure-g [x]
+  (let [y true] ; Overwrites type check
+    (if y
+      (t/ann-form (count x) t/Any) ; Type error: x could be any type
+      x)))
+
+;; Example nesting_condition
+;; success
+(t/ann nesting-condition-success-f [t/Any t/Any :-> t/Num])
+(defn nesting-condition-success-f [x y]
+  (if (if (number? x) (string? y) false)
+    (+ x (count y))
+    0))
+
+;; failure
+(t/ann nesting-condition-failure-f [t/Any t/Any :-> t/Num])
+(defn nesting-condition-failure-f [x y]
+  (if (if (number? x) (string? y) (string? y))
+    (t/ann-form (+ x (count y)) t/Num) ; Type error: x could be non-number
+    0))
+
+;; Example merge_with_union
+;; success
+(t/ann merge-with-union-success-f [t/Any :-> (t/U t/Str t/Num t/Any)])
+(defn merge-with-union-success-f [x]
+  (cond
```

Comment:
Move this `cond` into a let (like `merge-with-union-failure-f` does) and put the `return x` at the bottom.